src/pages/AppHub/AppHub.js

```diff
@@ -113,11 +114,19 @@ export const AppHub = () => {
         lazy: true,
         variables: {
             dhis_version: systemInfo.version,
+            // This is needed due to request caching - if the user visits the
+            // 'Custom apps' before the App Hub page, then `data.appHub` will
+            // be an array instead of an object as 'Custom apps' disables
+            // pagination. Moving `paging: true` to query params instead does
+            // not seem to resolve the issue.
+            paging: true,
```

Comment:
No idea if this is a bug with `useDataQuery` or if I'm doing something wrong :man_shrugging:

Reply:
cc @ismay

Reply:
I'll take a look 👀

Reply:
Ok, I see what's going wrong. See this PR: https://github.com/dhis2/app-management-app/pull/413. The problem is that the AppHub component uses this querykey:

```
const query = {
    appHub: {
        resource: 'appHub/v2/apps/',
        params: params => ({
            pageSize: 24,
            ...params,
        }),
    },
}
```

And CustomApps.js uses this query:

```
const appHubQuery = {
    appHub: {
        resource: 'appHub/v2/apps',
        params: ({ dhis_version }) => ({
            paging: false,
            dhis_version,
        }),
    },
}
```

Since react-query serializes the querykey to generate a cachekey these querykeys are considered identical (the function is disregarded as it can't be serialized). So when you load custom apps first you're adding the response there to the cache, and then when you load the app hub that component skips the loading stage and immediately receives the cached response that was originally meant for custom apps.

Reply:
In the linked PR, just appending a slash to the request in the apphub component fixes this. It's of course not a proper fix, but it demonstrates the issue.

Reply:
This is also why adding something unique to the variables for the AppHub query fixed this. It differentiated the query from the one for CustomApps.

Reply:
The best way to fix this within the current constraints of the app-runtime is probably to make sure the queryKeys are different by changing the resource name:


```javascript
const query = {
    appHub: {
        resource: 'appHub/v2/apps/',
        params: params => ({
            pageSize: 24,
            ...params,
        }),
    },
}
```

```javascript
const query = {
    customApps: {
        resource: 'appHub/v2/apps/',
        params: params => ({
            pageSize: 24,
            ...params,
        }),
    },
}
```

Reply:
https://dhis2.slack.com/archives/CLJSVA1RV/p1650378095993769